modules/canopennode/Kconfig

```diff
@@ -3,22 +3,56 @@
 # Copyright (c) 2019 Vestas Wind Systems A/S
 # SPDX-License-Identifier: Apache-2.0
 
-menuconfig CANOPEN
-	bool "CANopen protocol support"
+config ZEPHYR_CANOPENNODE_MODULE
+	bool
+
+config CANOPENNODE
+	bool "CANopenNode support"
 	depends on CAN
-	select CANOPENNODE
 	help
 	  Enable CANopen (EN 50325-4) (CiA 301) protocol
 	  support. Support is provided by the 3rd party CANopenNode
 	  protocol stack.
 
-if CANOPEN
+if CANOPENNODE
+
+# FIXME: Add support for named choice in the template
+#module = CANOPENNODE
+#module-str = CANopenNode
+#source "subsys/logging/Kconfig.template.log_config"
```

Comment:
For the template to work for this use-case, the `choice` at the top of the template needs to be named since it will included multiple times for module dependency resolution.

Our `kconfiglib.py` does not support macros in named choices and I have not had any success in trying to add support for it. For now, I have just copied over the template contents and converted the `choice` to a named choice, but this needs to be fixed.

Reply:
@tejlmand, @nashif: Any pointers on how to solve the above?

Reply:
Nevermind. @alexanderwachter and I decided to go for another solution, that can be used for logging in any future CANopen stacks as well.